pkg/loop/internal/relayerset/relayer.go

```diff
@@ -12,13 +12,19 @@ import (
 )
 
 type relayerClient struct {
+	*relayer.RelayerClient
 	log              logger.Logger
 	relayerSetClient *Client
```

Comment:
What is the difference between the three different things named `client`?

Reply:
This approach was abandoned because relayer server is not started and mounted by relayerset. So the relayer client initialised here would not be able to call NewContractReader, as there would be no corresponding server.

cc @cedric-cordenier

Reply:
Sorry I am not following. Can you explain in english what each of these three different things that all share the same generic name of "client" are actually doing? I think it will lead to better names, or at least something we can document in the code.

Reply:
There are just two clients here: a relayerClient and a relayerSetClient

A relayerClient is just a wrapper over the relayerSetClient + a relayID; this saves us having to register a relayer server every time we call RelayerSet.Get or RelayerSet.List since we defer fetching the relayer to the next call users make.

Reply:
There are three types in scope of this comment called "client":
- `relayerClient` or `relayerset.relayerClient`
- `relayer.RelayerClient`
- `Client` or `relayerset.Client`

Reply:
All of these type need documentation, or more intuitive names.

Reply:
Oh, was this removed?

Reply:
Yes, the original approach was removed because of the reason I mentioned above.